### PR TITLE
Support ID-Card signature/container generation by the backend

### PR DESCRIPTION
**Pull request by ykshatroff**
* Add some documentation to the IdCardManager.js methods
* Make the methods' API cleaner, use `resolve(value)` instead of storing all values on the instance
* Remove unnecessary parameter passing to the backend

### CODE REVIEW

File: IdCardManager.js [hunk 1]
```diff
@@ -1,125 +1,145 @@
...
-
-            window.hwcrypto.getCertificate(lParam).then((rCert) => {
-                this._cert = rCert;
-                resolve();
-            },
-
-            (err) => {
-                reject(err);
-            });
+            let lParam = { lang: this.language };
```
Comment (Jyrno42) [inline IdCardManager.js]: I guess this can be a const too

File: IdCardManager.js [hunk 2]
```diff
@@ -1,125 +1,145 @@
...
-
-            window.hwcrypto.sign(this._cert, {type: 'SHA-256', hex: hexData}, lParam)
-                .then((response) => {
-                    this._signature = response;
-                    resolve();
-
-                }, (err) => {
-                    reject(err);
-                });
+            let lParam = { lang: this.language };
```
Comment (Jyrno42) [inline IdCardManager.js]: ditto

**Review by Jyrno42 — CHANGES_REQUESTED** [review]

**Comment by Jyrno42** [comment]
I am also wondering if we could switch this to use `fetch` instead of superagent as fetch should mostly be available everywhere. This however can be done separately anyway.

File: IdCardManager.js (hunk 1 above)
Comment (ykshatroff) [reply IdCardManager.js]: I pretty much didn't touch it; but yeah makes sense to make it const :)

**Comment by ykshatroff** [comment]
@Jyrno42 

> fetch instead of superagent

I agree. I'll do a separate PR for this.

**Review by Jyrno42 — APPROVED** [review]

**Pull request merged**